### DISCUSSION TITLE
Support for Final cut?

### DISCUSSION THREAD

**Discussion by wilsonmcfee** [ntsc-rs/ntsc-rs#236] [Q&A]
hey! more of just a general question. Is there support for Final Cut? I'm really enjoying using the stand-alone version, but being able to keyframe edit parameters would give my videos so much more energy

Unless there's a way to do keyframes with the standalone? either way. 

Thanks!

**Comment by valadaptive** [comment]
I don't have any plans to add support for Final Cut Pro or Motion--I don't want to take on the work of supporting yet another API and can't afford to buy that software to test it out.

You could try DaVinci Resolve, which works on macOS and is free. It's definitely more cumbersome than being able to do it all in Final Cut, but it would let you keyframe things at least.

**Comment by NietzBlok** [comment]
I tried building it as a FXplug and I think I actually got it working. Will do some further testing and try to get it on GitHub. :). 

<img width="6240" height="3380" alt="Screenshot 2026-08-06 at 16 50 47" src="https://github.com/user-attachments/assets/c038a073-89c6-42c4-b05f-47f4bed7c79e" />

**Comment by wilsonmcfee** [comment]
Wow!!! This is awesome thank you so much. Cant wait to give this a try

Will

**Comment by NietzBlok** [comment]
If you want to have a go I've placed a build [here](https://www.dropbox.com/scl/fo/0raeajnvev0ys0ffjtslx/ALJ2VdLdrapC7qUwOWBQ5rs?rlkey=g4kbxgtrlr0xaxwe001dshqed&dl=0) for testing purposes. 

@valadaptive if you want to have a go I've put it on my GitHub. 

![NTSC-RS Icon](https://oostblok.media/NTSC-RS/NTSCRS.jpg "NTSC-RS")

**Reply by valadaptive** [reply]
Looks pretty good! I do have a couple questions:
- The processing order appears to be upside-down--the head-switching artifact *should* be at the bottom.
- Have you compared the performance of your Metal reimplementation to the original CPU implementation (single-threaded and multithreaded)? I've looked into using compute shaders, but the CPU version is already quite fast.
- In general, it may be easier to implement the plugin by just calling into the original ntsc-rs library instead of reimplementing the entire thing in Metal.

**Reply by valadaptive** [reply]
If you do want to pursue implementing the plugin in Rust, try pointing your Claude at the `objc2` crate and [its header translator](https://github.com/madsmtm/objc2/tree/main/crates/header-translator). I haven't actually checked myself since I don't own a copy of Final Cut, but it may be possible to write the *entire* thing in Rust (like I do with the After Effects and OpenFX plugins) if the FxPlug SDK has Objective-C headers.
